sway-fmt-v2/src/utils/comments.rs

```diff
@@ -0,0 +1,78 @@
+use anyhow::Result;
+use std::{cmp::Ordering, collections::BTreeMap, sync::Arc};
+use sway_parse::token::{lex_commented, Comment, CommentedTokenTree};
+
+/// Represents a span for the comments in a spesific file
+/// A stripped down version of sway-types::src::Span
+#[derive(PartialEq, Eq, Debug)]
+pub struct CommentSpan {
+    // The byte position in the string of the start of the span.
+    start: usize,
+    // The byte position in the string of the end of the span.
+    end: usize,
+}
+
+impl Ord for CommentSpan {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.start.cmp(&other.start)
```

Comment:
When it comes to ordering spans, I think the correct approach might be to reflect not just the order in which the `start` occurs, but also the order in which the spans would be visited during traversal (e.g. larger enclosing spans preceding inner spans.)

E.g. when comparing the spans `2..6` and `2..4`, the `2..6` span should come first as it encloses the `2..4` span. I think this would look something like:
```rust
match self.start.cmp(&other.start) {
    Ordering::Equal => other.end.cmp(&self.end),
    ord => ord,
}
```

Anyway, this is more relevant to the ordering of AST spans generally, and shouldn't be relevant to our case as I suppose comments cannot overlap :joy:

Reply:
I was actually thinking about this while writing that part but later on couldn't think of an example where comments would be overlapping as you pointed out 😄. But I will fix the ordering like this. Although it seems like overlapping won't be happening, if somehow it happens I feel like this will be really hard to debug.